### PR TITLE
Improve VIRTSER performance

### PR DESCRIPTION
**Pull request by xyzz**
## Description

Improve performance of virtser. Currently it consumes 1 byte at a time, which took for me over 30ms when sending 49 bytes. With a simple change, it can send 10x that in 30ms.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
This does affect steno, as it uses VIRTSER. 

Are you using this with Steno? Eg, has it been tested and verified that it doesn't break the steno implementation?

**Comment by xyzz** [comment]
No, I'm using it with a custom virtser_recv handler. I cannot check steno since I don't have the hardware. However, the chibios implementation of virtser does call it in a loop here: https://github.com/qmk/qmk_firmware/blob/96d4ba84c245066ae0ccd0f8216d7f11f80e5d98/tmk_core/protocol/chibios/usb_main.c#L835

**Comment by yanfali** [comment]
@germ can you verify this PR does not break steno, this would affect your boards more than others.

**Comment by germ** [comment]
I can test in about a week or so, I'm out of the country and didn't bring a laptop with me. I'll dump it in the Plover discord and see if someone can verify nothing breaks.

**Comment by drashna** [comment]
Awesome, thanks, @germ.

**Comment by yanfali** [comment]
> I can test in about a week or so, I'm out of the country and didn't bring a laptop with me. I'll dump it in the Plover discord and see if someone can verify nothing breaks.

Wait, they let you out of Canada?

**Comment by germ** [comment]
Hey! I'm not Australian or anything, don't get us confused :P

**Comment by antoniusf** [comment]
Hey! :wave: (I'm from the plover discord)

I just tested this with the georgi's default firmware, and everything seemed to work okay. I should say that I don't normally use a virtser-based protocol for steno, but I checked the behavior of this PR against master and didn't find any differences. As far as I know, the georgi uses the Gemini protocol, which means that TX Bolt is still untested, so if you want, I could try that as well. (On the other hand, the steno implementation doesn't even seem to use `virtser_recv`, only `virtser_send`, so it's probably fine anyway?)

File: tmk_core/protocol/lufa/lufa.c [hunk 1]
```diff
@@ -869,7 +869,7 @@ void virtser_recv(uint8_t c) {
 void virtser_task(void) {
     uint16_t count = CDC_Device_BytesReceived(&cdc_device);
     uint8_t  ch;
-    if (count) {
+    for (; count; --count) {
         ch = CDC_Device_ReceiveByte(&cdc_device);
```
Comment (fauxpark) [inline tmk_core/protocol/lufa/lufa.c]: Hm, possibly a small bug here: `CDC_Device_ReceiveByte()` will return `-1` if no data was received.
http://www.fourwalledcubicle.com/files/LUFA/Doc/170418/html/group___group___u_s_b_class_c_d_c_device.html#gaf02a74dffdcde55f4e522989e2ed49c1

**Pull request merged**